DOCS-2046: Update install and agent docs (#3243)
viamrobotics/docs: docs/fleet/provision.md

The configuration file must be stored at <file>/etc/viam-provisioning.json</file> and has the following format:

{{< tabs >}}
{{% tab name="Template" %}}

```json {class="line-numbers linkable-line-numbers"}
{
  "manufacturer": "<NAME>",
  "model": "NAME",
  "fragment_id": "<ID>",
  "hotspot_prefix": "<PREFIX>",
  "disable_dns_redirect": true,
  "hotspot_password": "<PASSWORD>",
  "roaming_mode": false,
  "offline_timeout": "0m00s",
  "user_timeout": "0m00s",
  "fallback_timeout": "0m00s"
}
```

{{% /tab %}}
{{% tab name="Example" %}}

```json {class="line-numbers linkable-line-numbers"}
{
  "manufacturer": "Skywalker",
  "model": "C-3PO",
  "fragment_id": "2567c87d-7aef-41bc-b82c-d363f9874663",
  "hotspot_prefix": "skywalker-setup",
  "disable_dns_redirect": true,
  "hotspot_password": "skywalker123",
  "roaming_mode": false,
  "offline_timeout": "3m30s",
  "user_timeout": "2m30s",
  "fallback_timeout": "15m"
}
```

This file configures some basic metadata, specifies a [fragment](/fleet/fragments/) to use to configure the machine, and provides the WiFi hotspot network name and password to use on startup.

> It also configures timeouts to control how long the Agent waits for a valid local WiFi network to come online before creating its hotspot network, and how long to keep the hotspot active before terminating it.

Reviewer: ```suggestion
It also configures timeouts to control how long the Viam Agent waits for a valid local WiFi network to come online before creating its hotspot network, and how long to keep the hotspot active before terminating it.
```

{{% /tab %}}
{{< /tabs >}}

{{% expand "Run additional provisioning steps with a pre-install script" %}}

When you install `viam-agent`, either manually using the commands above or automatically as part of your fleet's build and deploy process, you can choose to run a pre-install script to perform provisioning steps before deployment to the target machine.
For example, you could use the pre-install script to configure and deploy to an SD card or other image file, which you can then use as part of your fleet deployment process.
You can also use this method to generate a local tarball containing the configured deployment, which you could then deploy later, or through a different medium (such as automation, or as the basis for further customer-specific customization).

For more information, see [Pre-installed provisioning](https://github.com/viamrobotics/agent-provisioning#pre-installed-provisioning).
{{% /expand%}}

### Attributes

<!-- prettier-ignore -->
| Name       | Type   | Required? | Description |
| ---------- | ------ | --------- | ----------- |

> | `manufacturer` | string | Optional | Purely informative. May be displayed on captive portal and/or mobile app. Default: `"viam"`. |

Reviewer: ```suggestion
| `manufacturer` | string | Optional | Purely informative. May be displayed on captive portal and/or mobile app. Default: `"viam"`. |
```
do we use and/or? don't know how else i would put it but see this part is WIP

Author: ```suggestion
| `manufacturer` | string | Optional | Purely informative. May be displayed on captive portal or mobile app. Default: `"viam"`. |
```

By having the end customer set up the machine, the company:

- eliminates factory setup time
- allows for tailored configurations per customer as needed

{{< alert title="Support Notice" color="note" >}}

Provisioning works on Bullseye, Debian Bookworm or newer.
For Bullseye, the installation of `viam-agent` changes the network configuration to use NetworkManager.

{{< /alert >}}

## How an end user would use it

End users that receive a machine, use either a captive portal through a browser or a web or mobile app to complete the machine setup.

One option is to use the [Viam mobile app](/fleet/control/#control-interface-in-the-viam-mobile-app).
The Viam mobile app allows end users to create a new machine in the app, and `agent-provisioning` will then install `viam-server` and run it with a provided configuration.

To add your branding, you can build your own web or mobile app and use one of Viam's SDKs to connect to `viam-agent` and provision your machines:

- For mobile applications (existing or new) use the [Flutter SDK](https://flutter.viam.dev/viam_protos.provisioning.provisioning/ProvisioningServiceClient-class.html)

> - For web applications, use the [TypeScript SDK](https://github.com/viamrobotics/viam-typescript-sdk/blob/main/src/app/provisioning-client.ts).

Author: I don't know. I've been told the TypeScript SDK supports provisioning. But while reading your above comment I was also wondering if that would then be the captive portal it would replace... @ale7714 do you have more info on this?

1. When a machine that has `viam-agent` installed and `agent-provisioning` configured comes online, `agent-provisioning` creates a WiFi hotspot.

   - The [`agent-provisioning` configuration](#configuration) is at <file>/etc/viam-provisioning.json</file.
   - The [`viam-agent` configuration](/configure/agent/#configuration), if provided, is at <file>/etc/viam.json</file> alongside the rest of the machine configuration, including which fragments to use (if any).

1. The end user uses their mobile device or computer and connects to the WiFi hotspot.

   - By default, the hotspot network is named `viam-setup-HOSTNAME`, where `HOSTNAME` is replaced with the hostname of your machine.
     The WiFi password for this hotspot network is `viamsetup` by default.
     You can customize these values in the machine's or fragment's [`viam-agent` configuration](/configure/agent/#configuration).

1. If the end user has a provisioning web or mobile app, they should now open that app to complete setup.
   Alternatively, you will be redirected to a sign-in page.
   If you are using a laptop or are not redirected, try opening [http://viam.setup/](http://viam.setup/) in a browser.

> 1. In the browser or web or mobile app, the user will be prompted to provide the network information for the machine, that is the primary network for the machine.

Author: sorry, yeah I think I was tired. I don't know how this made sense to me yesterday to write

>    - If there is no machine configuration at <file>/etc/viam.json</file>, the provisioning app or captive portal will also require the user to paste an app configuration file.

Author: I think that's something we need to chat about further because we can't have a `viam-server` config and a `viam-server` configuration file. That's too confusing...

     - Select the part status dropdown to the right of your machine's name on the top of the page:
       {{<imgproc src="/build/micro-rdk/part-dropdown.png" resize="x600" style="max-width: 500px" declaredimensions=true alt="The part status dropdown of an offline machine.">}}
     - Click the copy icon underneath **Viam server configuration** to copy the `viam-server` app JSON configuration.
     - Paste the `viam-server` app config when prompted.

1. The machine will then disable the hotspot network and attempt to connected using the provided network information.
   If `viam-agent` cannot establish a connection using the provided network information, the machine will create the hotspot again and continue going through steps (2-5) until a connection is successfully established.
1. If the connection is successful, `viam-agent` installs `viam-server`.

   - `agent-provisioning` will use the provided network if it can connect, even if that network does not have internet access.
     Note that any features that require internet access will not function if the connected WiFi network is not connected to the internet.

>      If you want `agent-provisioning` to require that a WiFi network be connected to the internet in order to connect to it, enable roaming mode **and** do not provide a primary network.

Reviewer 2: ```suggestion
     If you want `agent-provisioning` to require that a WiFi network be connected to the internet in order to connect to it, enable roaming mode.
```
As noted elsewhere, when in roaming_mode, there is no "primary network."

> 1. `viam-agent` then starts `viam-server` with a provided configuration and the machine comes **live**.

Reviewer 2: ```suggestion
1. `viam-agent` then starts `viam-server` with the provided configuration and the machine becomes **live**.
```

   {{% /tab %}}

> {{% tab name="With pre-configured network (Roaming mode)" %}}

Reviewer 2: I'm not sure this tab makes sense at all. Adding (additional) networks via the agent-config (in the cloud) only applies AFTER provisioning is first completed. The config is stored in the cloud (App) so isn't downloaded to the machine until after the machine first connects anyway.

Reviewer 2: If a manufacturer wants to pre-install network configurations on an unprovisioned device, they should do so via the OS mechanisms directly. E.g. Add connections directly in NetworkManager.

Roaming mode allows you to specify multiple WiFi networks **in the machine's or fragment's [`viam-agent` configuration](/configure/agent/#configuration)**, each with their own priority, and `agent-provisioning` will try to connect to each specified network in order of priority from highest to lowest.

You must enable `roaming_mode` in the [`agent-provisioning` configuration](#configuration) to use roaming mode.

> Additionally, the end user must either **not** provide a primary network or there must be an issue with the connection to the primary network, for `agent-provisioning` to enter roaming mode.

Reviewer 2: As noted elsewhere, not true.

> `agent-provisioning` will enter roaming mode only when `roaming_mode` is enabled and no connection can be established to a primary network.

Reviewer 2: As noted above, not true. Roaming mode is enabled when "roaming_mode" is set to true. So this probably doesn't need to be clarified.

   - The [`agent-provisioning` configuration](#configuration) is at <file>/etc/viam-provisioning.json</file.
   - The [`viam-agent` configuration](/configure/agent/#configuration), is at <file>/etc/viam.json</file> alongside the rest of the machine configuration, including which fragments to use (if any).

1. In roaming mode, the machine attempts to connect to one or more pre-configured networks specified in the [`viam-agent` configuration](/configure/agent/#configuration) file in order of highest `priority`.

   - If none of the configured WiFi networks could be connected to, `agent-provisioning` will instead create its own WiFi hotspot and follow the flow for machines without pre-configured network.
     Consult the other tab for more information.
     You can configure how long `viam-agent` waits for connections and the timeout before it creates a hotspot in the [`agent-provisioning` configuration](#configuration).

1. If the connection is successful, `viam-agent` installs `viam-server`.
1. `viam-agent` then starts `viam-server` with a provided configuration and the machine comes **live**.

{{% /tab %}}
{{< /tabs >}}

## Configuration

> When you [install `viam-agent`](/configure/agent/#installation), provide a provisioning configuration file at <file>/etc/viam-provisioning.json</file> with the following format:

Reviewer 2: ```suggestion
When you [install `viam-agent`](/configure/agent/#installation), you may optionally provide a provisioning configuration file to customize the experience at <file>/etc/viam-provisioning.json</file> with the following format:
```

{
  "manufacturer": "Skywalker",
  "model": "C-3PO",
  "fragment_id": "2567c87d-7aef-41bc-b82c-d363f9874663",
  "hotspot_prefix": "skywalker-setup",
  "disable_dns_redirect": true,
  "hotspot_password": "skywalker123",
  "roaming_mode": false,
  "offline_timeout": "3m30s",
  "user_timeout": "2m30s",
  "fallback_timeout": "15m"
}
```

This file configures some basic metadata, specifies a [fragment](/fleet/fragments/) to use to configure the machine, and provides the WiFi hotspot network name and password to use on startup.
It also configures timeouts to control how long `viam-agent` waits for a valid local WiFi network to come online before creating its hotspot network, and how long to keep the hotspot active before terminating it.

{{% /tab %}}
{{< /tabs >}}

### Attributes

<!-- prettier-ignore -->
| Name       | Type   | Required? | Description |
| ---------- | ------ | --------- | ----------- |
| `manufacturer` | string | Optional | Purely informative. May be displayed on captive portal and/or mobile app. Default: `"viam"`. |
| `model` | string | Optional | Purely informative. May be displayed on captive portal and/or mobile app. Default: `"custom"`. |

> | `fragment_id` | string | Optional | The `fragment_id` of the fragment to configure machines with. If present, the Viam mobile app can pre-configure a robot for a user by using this [TODO: and otherwise it can't??]. |

Reviewer 2: Currently, we expect a fragment with our Mobile App experience. The mobile app is responsible for adding said fragment to the newly created machine's configuration.

| `hotspot_prefix` | string | Optional | Viam agent will prepend this to the hostname of the device append and use the resulting string for the provisioning hotspot SSID. Default: `"viam-setup"`. |
| `disable_dns_redirect` | boolean | Optional | By default, ALL DNS lookups using the provisioning hotspot will redirect to the device. This causes most phones/mobile devices to automatically redirect the user to the captive portal as a "sign in" screen. When disabled, only domains ending in .setup (ex: viam.setup) will be redirected. This generally avoids displaying the portal to users and is mainly used in conjunction with a mobile provisioning application workflow. Default: `false`. |

> | `hotspot_password` | string | Optional | The Wifi password for provisioning hotspot. Default: `"viamsetup"`. |

Reviewer 2: ```suggestion
| `hotspot_password` | string | Optional | The Wifi password for the provisioning hotspot. Default: `"viamsetup"`. |
```

> | `roaming_mode` | boolean | Optional | By default, the device will only attempt to connect to a single wifi network (the one with the highest priority), usually provided during initial provisioning/setup using the Viam app or the Viam mobile app. Wifi connection alone is enough to consider the device as "online" even if the global internet is not reachable. When enabled, the device will attempt connections to all configured networks, and only considers the device online if the internet is reachable. See [roaming mode](#how-an-end-user-would-use-it). Default: `false`. |

Reviewer 2: ```suggestion
| `roaming_mode` | boolean | Optional | By default, the device will only attempt to connect to a single wifi network (the one with the highest priority), usually provided during initial provisioning/setup using the Viam mobile app or captive web portal. Wifi connection alone is enough to consider the device as "online" even if the global internet is not reachable. When enabled, the device will instead attempt connections to all configured networks, and only considers the device online if the internet is reachable. See [roaming mode](#how-an-end-user-would-use-it). Default: `false`. |
```

| `offline_timeout` | boolean | Optional | Will only enter provisioning mode (hotspot) after being disconnected longer than this time. Useful on flaky connections, or when part of a system where the device may start quickly, but the wifi/router may take longer to be available. Default: `"2m"` (2 minutes). |

> | `user_timeout` | boolean | Optional | Amount of time before considering a user (using the provisioning portal using web or mobile app) idle, and resuming normal behavior. Used to avoid interrupting provisioning mode (for example for network tests/retries) when a user might be busy entering details. Default: `"5m"` (5 minutes). |

Reviewer 2: ```suggestion
| `user_timeout` | boolean | Optional | Amount of time before considering a user (using the captive web portal or mobile app) idle, and resuming normal behavior. Used to avoid interrupting provisioning mode (for example for network tests/retries) when a user might be busy entering details. Default: `"5m"` (5 minutes). |
```

| `fallback_timeout` | boolean | Optional | Provisioning mode will exit after this time, to allow other unmanaged (for example wired) or manually configured connections to be tried. Provisioning mode will restart if the connection/online status doesn't change. Default: `"10m"` (10 minutes). |

> | `networks` | boolean | Optional | The type of the network. See [Networks](/configure/agent/#networks). Default: `[]`. |

Reviewer 2: This is a complex structure, (same as in attributes.) It's not just "the type of the network." While it CAN be set here, it's advised/preferred that manufacturers instead directly enter wifi settings in the OS (e.g. directly in NetworkManager.)

Such a machine might use Viam to regularly capture and upload a stream of sensor readings, for example.
To parse the readings and provide tailored guidance to a ship's captain, the company writes their own proprietary application which includes live analytics and speech generation for conveying advice to the captain.

Using `agent-provisioning`, this company can ship their machines directly to customers with `viam-agent` installed.
When a customer sets up their machine, the provisioning subsystem installs `viam-server`.
By having the end customer set up the machine, the company:

- eliminates per-device setup and individualization at the factory
- allows for tailored configurations per customer as needed
- allows customer to provide their own WiFi credentials

{{< alert title="Support Notice" color="note" >}}

Provisioning is supported and tested only on Debian 11 (Bullseye), and 12 (Bookworm) but should work on most distros using NetworkManager v1.42 (or newer) as well.
For Bullseye, the installation of `viam-agent` changes the network configuration to use NetworkManager.

{{< /alert >}}

## End user experience

> End users receive a machine, and use either a captive web portal or mobile app to complete the machine setup.

Author: People using the typescript sdk... would their web apps replace the web portal and be run from the machine?

Reviewer 2: I think we cleared this up in slack, but for posterity. No, and that's the problem. We currently don't have a way to customize the browser-based experience in the web portal. Thus provisioning apps (regardless of SDK they use) need to be self-contained mobile apps (app store style) for the time being.

1. When a machine that has `viam-agent` installed and `agent-provisioning` configured comes online, `agent-provisioning` enters roaming mode:

   - The [`agent-provisioning` configuration](#configuration) is at <file>/etc/viam-provisioning.json</file.
   - The [`viam-agent` configuration](/configure/agent/#configuration), is at <file>/etc/viam.json</file> alongside the rest of the machine configuration, including which fragments to use (if any).

1. In roaming mode, the machine attempts to connect to one or more pre-configured networks specified in the [`viam-agent` configuration](/configure/agent/#configuration) file in order of highest `priority`.

   - If none of the configured WiFi networks could be connected to, `agent-provisioning` will instead create its own WiFi hotspot and follow the flow for machines without pre-configured network.
     Consult the other tab for more information.
     You can configure how long `viam-agent` waits for connections and the timeout before it creates a hotspot in the [`agent-provisioning` configuration](#configuration).

1. If the connection is successful, `viam-agent` installs `viam-server`.

> 1. `viam-agent` then starts `viam-server` with a provided configuration and the machine comes **live**.

Author: ```suggestion
1. `viam-agent` then starts `viam-server` with the provided configuration and the machine becomes **live**.
```

To add your branding, you can build your own mobile app and use the [Flutter SDK](https://flutter.viam.dev/viam_protos.provisioning.provisioning/ProvisioningServiceClient-class.html) or the [TypeScript SDK](https://github.com/viamrobotics/viam-typescript-sdk/blob/main/src/app/provisioning-client.ts) to connect to `viam-agent` and provision your machines.

If you are not using Flutter or TypeScript and would like to use provisioning, please [contact us](mailto:support@viam.com).

Regardless whether provisioning happens through the captive web portal or a mobile app, the process follows the same pattern:

{{<video webm_src="/platform/provisioning-demo.webm" mp4_src="/platform/provisioning-demo.mp4" alt="Using the Viam mobile app to provision a new machine with viam-agent." poster="/platform/provisioning-demo.jpg" class="alignright" max-width="400px" style="margin-left: 2rem">}}

1. When a machine that has `viam-agent` installed and `agent-provisioning` configured comes online, `agent-provisioning` creates a WiFi hotspot.

>    - The [`agent-provisioning` configuration](#configuration) is at <file>/etc/viam-provisioning.json</file.

Reviewer 2: ```suggestion
   - The [`agent-provisioning` configuration](#configuration) is at <file>/etc/viam-provisioning.json</file>.
```

Author: thx

>    - The [`viam-agent` configuration](/configure/agent/#configuration), if provided, is at <file>/etc/viam.json</file> alongside the rest of the machine configuration, including which fragments to use (if any).

Reviewer 2: This is not correct. The machine configuration (including fragments) is in the cloud/App. /etc/viam.json contains the machine credentials to connect TO the cloud.

> 1. The end user uses their mobile device or computer and connects to the WiFi hotspot.

Reviewer 2: If they're using the mobile app, they should open that first (while still on the internet) and follow the instructions there. (It needs to do stuff BEFORE connecting to the hotspot, and will then tell the user when to switch networks to the hotspot.)

   - By default, the hotspot network is named `viam-setup-HOSTNAME`, where `HOSTNAME` is replaced with the hostname of your machine.
     The WiFi password for this hotspot network is `viamsetup` by default.

>      You can customize these values in the machine's or fragment's [`viam-agent` configuration](/configure/agent/#configuration).

Reviewer 2: These are mostly meant to be customized in the provisioning configuration (/etc/viam-provisioning.json), not the Cloud configuration, as only that can be read/used when offline (e.g. before the user has first provisioned the machine.)

Author: that makes sense but why is it configurable in the agent config as well then?

Author: maybe it isn't

Author: Can you check the attributes in https://docs-test.viam.dev/3243/configure/agent/#agent-provisioning

1. If the end user has a provisioning mobile app, they should now open that app to complete setup.
   Alternatively, you will be redirected to a sign-in page.
   If you are using a laptop or are not redirected, try opening [http://viam.setup/](http://viam.setup/) in a browser.

1. In the captive web portal or mobile app, the user will be prompted to provide the network information for the machine.

>    - If there is no machine configuration at <file>/etc/viam.json</file>, the provisioning app or captive portal will also require the user to paste a machine credentials configuration file.

Reviewer 2: Only when using the web portal will the user needs to manually copy the machine credentials. The mobile app does this automatically.

(I think I pointed this out in my last review, but apparently all those comments are disconnected since the rebase...)

Author: sorry

     This is the JSON object which contains your machine part secret key and cloud app address, which your machine's `viam-server` instance needs to connect to the Viam app.

     To copy a machine credentials configuration file:

     - Navigate to your machine's page on [the Viam app](https://app.viam.com).
     - Select the part status dropdown to the right of your machine's name on the top of the page.
       {{<imgproc src="configure/machine-part-info.png" resize="500x" declaredimensions=true alt="Restart button on the machine part info dropdown">}}
     - Click the copy icon next to **Viam server config** to copy the machine credentials configuration file.
     - Paste the `viam-server` app config when prompted.

> 1. The machine will then disable the hotspot network and attempt to connected using the provided network information.

Reviewer 2: ```suggestion
1. The machine will then disable the hotspot network and attempt to connect using the provided network information.
```

   If `viam-agent` cannot establish a connection using the provided network information, the machine will create the hotspot again and continue going through steps (2-5) until a connection is successfully established.
1. If the connection is successful, `viam-agent` installs `viam-server`.

   - `agent-provisioning` will use the provided network if it can connect, even if that network does not have internet access.
     Note that any features that require internet access will not function if the connected WiFi network is not connected to the internet.
     If you want `agent-provisioning` to require that a WiFi network be connected to the internet in order to connect to it, enable roaming mode.

1. `viam-agent` then starts `viam-server` with the provided configuration and the machine becomes **live**.

## Configuration

When you [install `viam-agent`](/configure/agent/#installation), you may optionally provide a provisioning configuration file to customize the experience at <file>/etc/viam-provisioning.json</file> with the following format:

{{< tabs >}}
{{% tab name="Template" %}}

```json {class="line-numbers linkable-line-numbers"}
{
  "manufacturer": "<NAME>",

>   "model": "NAME",

Reviewer 2: ```suggestion
  "model": "<NAME>",
```

  "roaming_mode": false,
  "offline_timeout": "3m30s",
  "user_timeout": "2m30s",
  "fallback_timeout": "15m"
}
```

This file configures some basic metadata, specifies a [fragment](/fleet/fragments/) to use to configure the machine, and provides the WiFi hotspot network name and password to use on startup.
It also configures timeouts to control how long `viam-agent` waits for a valid local WiFi network to come online before creating its hotspot network, and how long to keep the hotspot active before terminating it.

{{% /tab %}}
{{< /tabs >}}

### Attributes

<!-- prettier-ignore -->
| Name       | Type   | Required? | Description |
| ---------- | ------ | --------- | ----------- |
| `manufacturer` | string | Optional | Purely informative. May be displayed on captive portal or provisioning app. Default: `"viam"`. |
| `model` | string | Optional | Purely informative. May be displayed on captive portal or provisioning app. Default: `"custom"`. |
| `fragment_id` | string | Optional | The `fragment_id` of the fragment to configure machines with. Required when using the Viam mobile app for provisioning. The Viam mobile app uses the fragment to configure the machine. |

> | `hotspot_prefix` | string | Optional | Viam agent will prepend this to the hostname of the device append and use the resulting string for the provisioning hotspot SSID. Default: `"viam-setup"`. |

Reviewer 2: ```suggestion
| `hotspot_prefix` | string | Optional | Viam agent will prepend this to the hostname of the device and use the resulting string for the provisioning hotspot SSID. Default: `"viam-setup"`. |
```

| `offline_timeout` | boolean | Optional | Will only enter provisioning mode (hotspot) after being disconnected longer than this time. Useful on flaky connections, or when part of a system where the device may start quickly, but the wifi/router may take longer to be available. Default: `"2m"` (2 minutes). |
| `user_timeout` | boolean | Optional | Amount of time before considering a user (using the captive web portal or provisioning app) idle, and resuming normal behavior. Used to avoid interrupting provisioning mode (for example for network tests/retries) when a user might be busy entering details. Default: `"5m"` (5 minutes). |
| `fallback_timeout` | boolean | Optional | Provisioning mode will exit after this time, to allow other unmanaged (for example wired) or manually configured connections to be tried. Provisioning mode will restart if the connection/online status doesn't change. Default: `"10m"` (10 minutes). |

> | `networks` | boolean | Optional | Add additional network the machine can connect to for provisioning. We recommend that you add WiFi settings in the operating system (for example, directly in NetworkManager). See [Networks](/configure/agent/#networks). Default: `[]`. |

Reviewer 2: ```suggestion
| `networks` | boolean | Optional | Add additional networks the machine can connect to for provisioning. We recommend that you add WiFi settings in the operating system (for example, directly in NetworkManager) rather than in this file, or in the corresponding machine config in App if networks aren't needed until after initial provisioning. See [Networks](/configure/agent/#networks). Default: `[]`. |
```

For Bullseye, the installation of `viam-agent` changes the network configuration to use NetworkManager.

{{< /alert >}}

## End user experience

End users receive a machine, and use either a captive web portal or mobile app to complete the machine setup.

One option is to use the [Viam mobile app](/fleet/control/#control-interface-in-the-viam-mobile-app).
The Viam mobile app allows end users to create a new machine in the app, and `agent-provisioning` will then install `viam-server` and run it with a provided configuration.

To add your branding, you can build your own mobile app and use the [Flutter SDK](https://flutter.viam.dev/viam_protos.provisioning.provisioning/ProvisioningServiceClient-class.html) or the [TypeScript SDK](https://github.com/viamrobotics/viam-typescript-sdk/blob/main/src/app/provisioning-client.ts) to connect to `viam-agent` and provision your machines.

If you are not using Flutter or TypeScript and would like to use provisioning, please [contact us](mailto:support@viam.com).

This is the general process for provisioning depending on whether you are using a captive web portal or a mobile app:

{{< tabs >}}

> {{% tab name="Mobile app" min-height="703px" %}}

Reviewer 2: I'm unclear on the mobile app flow. Previously (when I last used it) you could only use it to provision NEW machines. If that's still the case, then the reference below to where configuration is for machines that already exist is pointless (but also unlikely to be harmful.) I'm fine with it either way, so I'll leave this section for @ale7714 to sign off on.

Author: I'll remove it.

Author: for now, will actually test this in the next week

If you are not using Flutter or TypeScript and would like to use provisioning, please [contact us](mailto:support@viam.com).

This is the general process for provisioning depending on whether you are using a captive web portal or a mobile app:

{{< tabs >}}
{{% tab name="Mobile app" min-height="703px" %}}

{{<video webm_src="/platform/provisioning-demo.webm" mp4_src="/platform/provisioning-demo.mp4" alt="Using the Viam mobile app to provision a new machine with viam-agent." poster="/platform/provisioning-demo.jpg" class="alignright" max-width="400px" style="margin-left: 2rem">}}

1. If the provisioning happens with a mobile app, open the app and follow any instructions there until the app directs you to turn on the machine.

   - If you are using the Viam mobile app, create a new machine and then follow the instructions.

1. When you power on the machine that has `viam-agent` installed and `agent-provisioning` configured, `agent-provisioning` creates a WiFi hotspot.

   - The [`agent-provisioning` configuration](#configuration) is at <file>/etc/viam-provisioning.json</file>.

>    - If a machine already exists, a machine credentials configuration file, if provided, is at <file>/etc/viam.json</file>.

Author: ```suggestion
```
